browser/net/brave_site_hacks_network_delegate_helper.cc

```diff
@@ -112,6 +126,36 @@ bool IsBlockTwitterSiteHack(net::URLRequest* request,
   return false;
 }
 
+int ApplyPotentialReferrerBlock(net::URLRequest* request,
+    const ResponseCallback& next_callback,
+    net::HttpRequestHeaders* headers) {
+  DCHECK_CURRENTLY_ON(BrowserThread::IO);
+  GURL target_origin = GURL(request->url()).GetOrigin();
+  GURL tab_origin = request->site_for_cookies().GetOrigin();
+  bool allow_referrers = brave_shields::IsAllowContentSettingFromIO(
+      request, tab_origin, tab_origin, CONTENT_SETTINGS_TYPE_PLUGINS,
+      brave_shields::kReferrers);
+  bool shields_up = brave_shields::IsAllowContentSettingFromIO(
+      request, tab_origin, GURL(), CONTENT_SETTINGS_TYPE_PLUGINS,
+      brave_shields::kBraveShields);
+  std::string referrer;
+  headers->GetHeader(kRefererHeader, &referrer);
+  if (headers->GetHeader(kRefererHeader, &referrer) &&
+      !IsWhitelistedReferrer(tab_origin, target_origin) &&
+      !allow_referrers &&
+      shields_up &&
+      !SameDomainOrHost(target_origin, GURL(referrer),
+          INCLUDE_PRIVATE_REGISTRIES)) {
+    auto referrer = Referrer::SanitizeForRequest(target_origin,
+        Referrer(target_origin,
+            request->referrer_policy() == net::URLRequest::NO_REFERRER ?
```

Comment:
I don't think this is the correct thing to do here. `SanitizeForRequest` should handle this if you just pass the `referrer_policy`